platforms/jvm/testing-jvm/src/integTest/groovy/org/gradle/testing/nonclassbased/NonClassBasedTestingIntegrationTest.groovy

```diff
@@ -297,7 +297,211 @@ class NonClassBasedTestingIntegrationTest extends AbstractNonClassBasedTestingIn
         containsLine(result.getOutput(), "FINISH [Test SomeTestSpec.rbt : foo()] [SomeTestSpec.rbt : foo] [SKIPPED] [1] [null]")
         containsLine(result.getOutput(), "START [Test SomeTestSpec.rbt : bar()] [SomeTestSpec.rbt : bar]")
         containsLine(result.getOutput(), "FINISH [Test SomeTestSpec.rbt : bar()] [SomeTestSpec.rbt : bar] [SKIPPED] [1] [null]")
-        containsLine(result.getOutput(), "START [Test subSomeOtherTestSpec.rbt : other()] [subSomeOtherTestSpec.rbt : other]")
-        containsLine(result.getOutput(), "FINISH [Test subSomeOtherTestSpec.rbt : other()] [subSomeOtherTestSpec.rbt : other] [SKIPPED] [1] [null]")
+        containsLine(result.getOutput(), "START [Test SomeOtherTestSpec.rbt : other()] [SomeOtherTestSpec.rbt : other]")
+        containsLine(result.getOutput(), "FINISH [Test SomeOtherTestSpec.rbt : other()] [SomeOtherTestSpec.rbt : other] [SKIPPED] [1] [null]")
     }
+
+    def "resource-based test engine can exclude test definitions"() {
+        given:
+        buildFile << """
+            plugins {
+                id 'java-library'
+            }
+
+            ${mavenCentralRepository()}
+
+            testing.suites.test {
+                ${enableEngineForSuite()}
+
+                targets.all {
+                    testTask.configure {
+                        testDefinitionDirs.from("$DEFAULT_DEFINITIONS_LOCATION")
+
+                        filter {
+                            excludeTestsMatching "*SomeTestSpec*"
+                        }
+                    }
+                }
+            }
+        """
+
+        writeTestDefinitions()
+
+        when:
+        succeeds("test", "--info")
+
+        then:
+        outputDoesNotContain("INFO: Executing resource-based test: Test[file=SomeTestSpec.rbt, name=foo]")
+        outputDoesNotContain("INFO: Executing resource-based test: Test[file=SomeTestSpec.rbt, name=bar]")
+        outputContains("INFO: Executing resource-based test: Test[file=SomeOtherTestSpec.rbt, name=other]")
```

Comment:
Maybe not as part of this, but we should not be using console output to assert that certain tests have executed.  We should be checking the reports.  `outputDoesNotContain` and `outputContains` are not reliable because `outputDoesNotContain` will allow typos to pass through and `outputContains` will allow extra tests to execute.